src/js/letters/components/Address.jsx

```diff
@@ -69,75 +61,104 @@ class Address extends React.Component {
   }
 
   render() {
+    const addressType = this.props.value.type;
+
+    // Depending on the type of address, the fields for state and city
+    // are called different things. Set new variables for the name of
+    // the appropriate city and state field, and use those below for
+    // the field components.
+    let cityField;
+    let stateField;
+    if (addressType === 'MILITARY') {
+      cityField = 'militaryPostOfficeTypeCode';
```

Comment:
Hm...now that you point this out, it's wrong. We got clarity on requirements since this was written: https://github.com/department-of-veterans-affairs/vets.gov-team/issues/4110#issuecomment-328640426

Reply:
Hm sorry missing what's wrong here @raquelromano-gov? I see `militaryStateCode` corresponds with the `state` field, so it would seem `militaryPostOfficeTypeCode` corresponds to the city field (at least this is what we have done on all other forms, where the options of "APO", "FPO", and "DPO" stand in for the city). The logic is a bit odd, how `militaryStateCode` is asked first. This would change the order of our inputs, and change the conditional logic of when inputs are displayed or hidden. This makes things more complicated without knowing the type of address first. Might be worth talking this through with @goldenmeanie tomorrow.

Reply:
Oh, OK--@goldenmeanie and I had discussed that people with overseas military addresses are used to picking US as their country and looking for AA/AE/AP in a list of states. But I didn't catch that they also know to put APO/FPO/DPO in as the city type (I thought we'd be adding a conditional dropdown menu explicitly for the military post office code if a military state were chosen. So if that's what the research shows, that sounds fine to me.